library/src/window_impl.cxx

```diff
@@ -139,54 +156,120 @@ class window_impl::internals
+  this->Internals->GetProcAddress = getProcAddress;
   if (type == Type::NONE)
   {
     this->Internals->RenWin = vtkSmartPointer<vtkF3DNoRenderWindow>::New();
   }
   else if (type == Type::EXTERNAL)
   {
 #if F3D_MODULE_EXTERNAL_RENDERING
-    this->Internals->RenWin = vtkSmartPointer<vtkExternalOpenGLRenderWindow>::New();
+    vtkNew<vtkExternalOpenGLRenderWindow> extWin;
+    extWin->AutomaticWindowPositionAndResizeOff();
+    this->Internals->RenWin = extWin;
 #else
     throw engine::no_window_exception(
       "Window type is external but F3D_MODULE_EXTERNAL_RENDERING is not enabled");
 #endif
   }
-  else
+  else if (type == Type::EGL)
   {
-    this->Internals->RenWin = vtkSmartPointer<vtkRenderWindow>::New();
-    this->Internals->RenWin->SetOffScreenRendering(type == Type::NATIVE_OFFSCREEN);
-    this->Internals->RenWin->SetMultiSamples(0); // Disable hardware antialiasing
-
+#if defined(VTK_OPENGL_HAS_EGL) && VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 3, 20240914)
+    this->Internals->RenWin = vtkSmartPointer<vtkEGLRenderWindow>::New();
 #ifdef __ANDROID__
     // Since F3D_MODULE_EXTERNAL_RENDERING is not supported on Android yet, we need to call
     // this workaround. It makes vtkEGLRenderWindow external if WindowInfo is not nullptr.
     this->Internals->RenWin->SetWindowInfo("jni");
+#endif
+#else
+    throw engine::no_window_exception("Window type is EGL but VTK EGL support is not enabled");
 #endif
   }
+  else if (type == Type::OSMESA)
+  {
+#if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 3, 20240914)
+    this->Internals->RenWin = vtkSmartPointer<vtkOSOpenGLRenderWindow>::New();
+#else
+    throw engine::no_window_exception(
+      "Window type is OSMesa but VTK OSMesa support is not enabled");
+#endif
+  }
+  else if (type == Type::GLX)
+  {
+#if defined(VTK_USE_X) && VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 3, 20240914)
+    this->Internals->RenWin = vtkSmartPointer<vtkXOpenGLRenderWindow>::New();
+#else
+    throw engine::no_window_exception("Window type is GLX but VTK GLX support is not enabled");
+#endif
+  }
+  else if (type == Type::WGL)
+  {
+#ifdef _WIN32
+    this->Internals->RenWin = vtkSmartPointer<vtkWin32OpenGLRenderWindow>::New();
+#endif
+  }
+  else if (!type.has_value())
+  {
+    // rely on VTK logic
+    this->Internals->RenWin = vtkSmartPointer<vtkRenderWindow>::New();
+  }
+
+  assert(this->Internals->RenWin != nullptr);
+
+#if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 3, 20240914)
+  vtkOpenGLRenderWindow* oglRenWin = vtkOpenGLRenderWindow::SafeDownCast(this->Internals->RenWin);
+  if (oglRenWin)
+  {
+    if (this->Internals->GetProcAddress)
+    {
+      oglRenWin->SetOpenGLSymbolLoader(&internals::SymbolLoader, &this->Internals->GetProcAddress);
+    }
+#if F3D_MODULE_EXTERNAL_RENDERING
+    if (oglRenWin->IsA("vtkExternalOpenGLRenderWindow"))
+    {
+      // We need to call vtkOpenGLRenderWindow function because vtkGenericOpenGLRenderWindow is
+      // reimplementing it incorrectly
```

Comment:
should an issue be open on VTK ?

Reply:
I don't know the details, I just put a comment based on the discussion on Discord.  
I understand the issue, but I don't know the generic window well enough to know if it's a bug or a limitation.

Reply:
fair enough